src/ConnectedMode/Migration/MSBuildFileProvider.cs

```diff
@@ -31,18 +38,111 @@ namespace SonarLint.VisualStudio.ConnectedMode.Migration
     [PartCreationPolicy(CreationPolicy.NonShared)]
     internal class MSBuildFileProvider : IFileProvider
     {
+        private readonly IServiceProvider serviceProvider;
         private readonly ILogger logger;
+        private readonly IThreadHandling threadHandling;
+        private readonly IFileSystem fileSystem;
+
+        internal static readonly string[] FileSearchPatterns = new string[] {
+            "*.ruleset",
+            "*.props",
+            "*.targets",
+            "*.csproj",
+            "*.vbproj"
+        };
+
+        internal static readonly string[] ExcludedDirectores = new string[]
+        {
+            "\\bin\\",
+            "\\obj\\"
+        };
 
         [ImportingConstructor]
-        public MSBuildFileProvider(ILogger logger)
+        public MSBuildFileProvider(
+            [Import(typeof(SVsServiceProvider))] IServiceProvider serviceProvider,
+            ILogger logger,
+            IThreadHandling threadHandling)
+            : this(serviceProvider, logger, threadHandling, new FileSystem())
         {
+        }
+
+        internal /* for testing */ MSBuildFileProvider(
+            [Import(typeof(SVsServiceProvider))] IServiceProvider serviceProvider,
+            ILogger logger,
+            IThreadHandling threadHandling,
+            IFileSystem fileSystem)
+        {
+            this.serviceProvider = serviceProvider;
             this.logger = logger;
+            this.threadHandling = threadHandling;
+            this.fileSystem = fileSystem;
+        }
+
+        public async Task<IEnumerable<string>> GetFilesAsync(CancellationToken token)
+        {
+            var solutionDir = await GetSolutionDirectoryAsync();
+            if (solutionDir == null)
+            {
+                LogVerbose("Unable to locate the solution folder.");
+                return Enumerable.Empty<string>();
+            }
+
+            // Make sure we are doing the disk search on a background thread
+            await threadHandling.SwitchToBackgroundThread();
```

Comment:
We'll need to look at this as part of the end-to-end migration threading, starting from the wizard down.